libethereum/BlockChain.cpp

```diff
@@ -864,6 +864,8 @@ ImportRoute BlockChain::import(VerifiedBlockRef const& _block, OverlayDB const&
 		{
 			newLastBlockHash = _block.info.hash();
 			newLastBlockNumber = (unsigned)_block.info.number();
+			if (m_onBlockImport)
+				m_onBlockImport(_block.info);
```

Comment:
Would it be better to call this callback after we have written everything to the database (i.e. at the end of this method)?